src/spec/authentication_service.md

```diff
@@ -34,31 +30,112 @@ The AS generally keeps the following state
         * **Encryption ratchet key:** Symmetric key used to derive queue encryption keys.
       * **Current sequence number:** The current message sequence number.
       * **Queued messages:** A sequence of ciphertexts containing the messages in the queue. Each incoming message is [encrypted](./queuing_service/queue_encryption.md) and is assigned the current sequence number, after which the current sequence number is incremented.
+* **Ephemeral OPAQUE DB:** An in-memory database that stores user-name or client-id indexed entries
+  * **Client credential:** Credential of the client that is being added to an existing user entry, or the initial client of a new user entry.
+  * **Optional OPAQUE state:** OPAQUE server state, only present in case of the OPAQUE online AKE flow, i.e. if a client performs 2FA, but not during an OPAQUE setup.
+* **AS credential key material:** Credentials that can be retrieved by clients, as well as the private key material used by the AS to sign client [CSRs](authentication_service/credentials.md#client-credential-signing-requests).
+* **AS OPAQUE key material:** OPRF seed, server public key and server private key as required for the server to perform OPAQUE registration a login flows with clients.
 
 ## Authentication
 
-The AS uses the user's IdP to authenticate the user. Some endpoints require simple authentication, while for others, the AS requires MFA authentication via the IdP.
+There are three modes of authentication for endpoints on the AS.
 
-## User account registration
+* **None:** For endpoints that are meant to be publicly accessible, e.g. user account registration
+* **Client Credential:** A signature over a time stamp using the signature key in the calling client's ClientCredential.
+* **Client Password:** An OPAQUE login flow.
+* **Client 2FA:** The same as **Client**, but additionally performing an [OPAQUE](https://datatracker.ietf.org/doc/draft-irtf-cfrg-opaque/) login flow
 
-The user registration functionality delegates authentication to an OpenID connect-capable identity provider. Note, that the user's chosen user name is part of the `client_csr`.
+## Initiate 2FA operation
+
+The client has to query this endpoint before they can query an endpoint that requires **Client 2FA** authentication. Note that this endpoint is not meant to be used with endpoints that require **Client Password** authentication such as the client addition endpoint. Similarly, this endpoint is not meant to be used to set up 2FA (see [here](./authentication_service.md#initialize-user-registration) instead).
```

Comment:
```suggestion
The client has to query this endpoint before it can query an endpoint that requires **Client 2FA** authentication. Note that this endpoint is not meant to be used with endpoints that require **Client Password** authentication such as the client addition endpoint. Similarly, this endpoint is not meant to be used to set up 2FA (see [here](./authentication_service.md#initialize-user-registration) instead).
```